test/FunctionalTests/Microsoft.OData.Edm.Tests/Csdl/CsdlWriterTests.TargetPath.cs

```diff
@@ -0,0 +1,448 @@
+                      "<Property Name=\"Id\" Type=\"Edm.Int32\" Nullable=\"false\" />" +
+                      "<Property Name=\"Name\" Type=\"Edm.String\" Nullable=\"false\" />" +
+                    "</EntityType>" +
+                    "<Term Name=\"MyTerm\" Type=\"Edm.String\" />" +
+                    "<EntityContainer Name=\"Default\">" +
+                       "<EntitySet Name=\"Customers\" EntityType=\"NS.Customer\" />" +
+                    "</EntityContainer>" +
+                    "<Annotations Target=\"NS.Default/Customers/Name\">" +
+                      "<Annotation Term=\"NS.MyTerm\" String=\"Name OutOfLine MyTerm Value\" />" +
+                    "</Annotations>" +
+                  "</Schema>" +
+                "</edmx:DataServices>" +
+              "</edmx:Edmx>");
+
+            // Act & Assert for JSON
+            WriteAndVerifyJson(model, @"{
+  ""$Version"": ""4.0"",
+  ""$EntityContainer"": ""NS.Default"",
+  ""NS"": {
+    ""Customer"": {
+      ""$Kind"": ""EntityType"",
+      ""$Key"": [
+        ""Id""
+      ],
+      ""Id"": {
+        ""$Type"": ""Edm.Int32""
+      },
+      ""Name"": {}
+    },
+    ""MyTerm"": {
+      ""$Kind"": ""Term"",
+      ""$Nullable"": true
+    },
+    ""Default"": {
+      ""$Kind"": ""EntityContainer"",
+      ""Customers"": {
+        ""$Collection"": true,
+        ""$Type"": ""NS.Customer""
+      }
+    },
+    ""$Annotations"": {
+      ""NS.Default/Customers/Name"": {
+        ""@NS.MyTerm"": ""Name OutOfLine MyTerm Value""
+      }
+    }
+  }
+}");
+        }
+
+        [Fact]
+        public void WriteInlineAnnotationForTargetPathThrows()
+        {
+            // Arrange
+            EdmModel model = new EdmModel();
+            EdmEntityType customer = new EdmEntityType("NS", "Customer");
+            customer.AddKeys(customer.AddStructuralProperty("Id", EdmCoreModel.Instance.GetInt32(false)));
+            customer.AddStructuralProperty("Name", EdmPrimitiveTypeKind.String, isNullable: false);
+            model.AddElement(customer);
+
+            EdmEntityContainer container = new EdmEntityContainer("NS", "Default");
+            EdmEntitySet entitySet = new EdmEntitySet(container, "Customers", customer);
+            container.AddElement(entitySet);
+            model.AddElement(container);
+
+            IEdmProperty nameProperty = customer.DeclaredProperties.Where(x => x.Name == "Name").FirstOrDefault();
+
+            EdmTargetPath targetPath = new EdmTargetPath(container, entitySet, nameProperty);
+
+            EdmTerm term = new EdmTerm("NS", "MyTerm", EdmCoreModel.Instance.GetString(true));
+            model.AddElement(term);
+            EdmVocabularyAnnotation annotation = new EdmVocabularyAnnotation(targetPath, term, new EdmStringConstant("Name OutOfLine MyTerm Value"));
+
+            // Act & Assert
+            Action action = () => annotation.SetSerializationLocation(model, EdmVocabularyAnnotationSerializationLocation.Inline);

```

Comment:
>SetSerializationLocation [](http://example.com/codeflow?start=45&length=24)

Since this is failing test and all you need is the 'interfaces', it's better to use Moq.

Mock<IEdmTargetPath> targetPath = new Mock<IEdmTargetPath>();
Mock<IEdmModel> model = new Mock<IEdmModel>();
Mock<IEdmVocabularyAnnotation> annotation= new Mock<IEdmVocalbuarlyAnnotation>();
annotation.Setup(a => a.Target).Returns(targetPath.Object);
....